Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1

```diff
@@ -0,0 +1,60 @@
+function ShouldBeInResourceGroup ($ActualValue, [string]$ExpectedValue, [switch] $Negate, [string] $Because) {
+  <#
+    .SYNOPSIS
+      Custom Assertion function to check resource's resource group.
+
+    .DESCRIPTION
+      BeInResourceGroup is a custom assertion that checks the resource group of a resource.
+      It can be used when writing Pester tests.
+
+    .EXAMPLE
+      $result = Confirm-AzBPContainerRegistry -ResourceGroupName $rgName
+
+      $result | Should -BeInResourceGroup 'testrg'
+
+    .EXAMPLE
+      $result = Confirm-AzBPContainerRegistry -ResourceGroupName $rgName
+
+      $result | Should -Not -BeInResourceGroup 'testrg2'
+
+    .INPUTS
+      ConfirmResult
+      System.Switch
+      System.String
+
+    .OUTPUTS
+      PSCustomObject
+  #>
+  if ($null -eq $ActualValue){
+    [bool] $succeeded = $false
+
+    if ($Negate) { $succeeded = -not $succeeded }
+    $failureMessage = "Confirm result is null or empty."
+  }
+  else {
+    $resourceGroupName = $ActualValue.ResourceDetails.ResourceGroupName
+    [bool] $succeeded = $resourceGroupName -eq $ExpectedValue
+    if ($Negate) { $succeeded = -not $succeeded }
+
+    if (-not $succeeded) {
+        if ($Negate) {
+          $failureMessage = "Resource not in resource group. This is expected for $ExpectedValue"
```

Comment:
I don't think that this message is correct. The code can only get here if the resource is *in* the resource group, which is a failure.

`$succeeded` has already been normalized depending on whether it is negated or not, so if the code gets into the `if (-not $succeeded)` then it has failed regardless of whether there has been negation. Negation just determines why it failed.

This applies to all three of the custom assertions.